modules/azure/compute.go

```diff
@@ -8,97 +8,392 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// GetVirtualMachineClientE is a helper function that will setup an Azure Virtual Machine client on your behalf.
-func GetVirtualMachineClientE(subscriptionID string) (*compute.VirtualMachinesClient, error) {
-	// Validate Azure subscription ID
-	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+// VirtualMachineExists indicates whether the specifcied Azure Virtual Machine exists.
+// This function would fail the test if there is an error.
+func VirtualMachineExists(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) bool {
+	exists, err := VirtualMachineExistsE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// VirtualMachineExistsE indicates whether the specifcied Azure Virtual Machine exists.
+func VirtualMachineExistsE(vmName string, resGroupName string, subscriptionID string) (bool, error) {
+	// Get VM Object
+	_, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetVirtualMachineNics gets a list of Network Interface names for a specifcied Azure Virtual Machine.
+// This function would fail the test if there is an error.
+func GetVirtualMachineNics(t testing.TestingT, vmName string, resGroupName string, subscriptionID string) []string {
+	nicList, err := GetVirtualMachineNicsE(vmName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+
+	return nicList
+}
+
+// GetVirtualMachineNicsE gets a list of Network Interface names for a specified Azure Virtual Machine.
+func GetVirtualMachineNicsE(vmName string, resGroupName string, subscriptionID string) ([]string, error) {
+
+	// Get VM Object
+	vm, err := GetVirtualMachineE(vmName, resGroupName, subscriptionID)
 	if err != nil {
 		return nil, err
 	}
 
-	// Create a VM client
-	vmClient := compute.NewVirtualMachinesClient(subscriptionID)
+	vmNICs := *vm.NetworkProfile.NetworkInterfaces
```

Comment:
Do you need to check this isn't `nil`? E.g., If there are no NICs attached? It's really important to avoid panics in Terratest.

Reply:
One NIC is required or you get the Terraform error below, so technically the length check is not necessary either. I also confirmed that once created, at leas one NIC is required and cannot be removed via the UI and the attached NIC cannot be deleted because it is a dependency.

Error: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="VirtualMachineMustHaveAtLeastOneNetworkInterface" Message="Virtual machine vm-resource must have at least one network interface." Details=[]

Added comment to support the absence of the `nil` check.